libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewListener.kt

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.Style
+import com.mapbox.navigation.dropin.component.camera.TargetCameraMode
+import com.mapbox.navigation.dropin.component.destination.Destination
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+
+/**
+ * Interface definition for the NavigationView.
+ */
+abstract class NavigationViewListener {
+    /**
+     * Called when [Destination] has changed. Might be called multiple times once
+     * [Destination.features] are updated.
+     *
+     * @param destination New destination value.
+     */
+    open fun onDestinationChanged(destination: Destination?) = Unit
+
+    /**
+     * Called when [NavigationState] has changed.
+     *
+     * @param state New navigation state value.
+     */
+    open fun onNavigationStateChanged(state: NavigationState) = Unit
```

Comment:
Exposing `NavigationState` iterable `sealed class` prevents us from adding additional states (like search or stop management), if we wanted to expose these as states in the future. Is there a way for us to provide the same access to relevant information (of when a state is set) without having to expose the iterable? Possibly proposal from https://github.com/mapbox/mapbox-navigation-android/pull/5922#issuecomment-1155497080 covers that as well.

Reply:
I guess the same applies to `TargetCameraMode` exposed via `onCameraModeChanged(mode: TargetCameraMode)`.
What if we substitute those states and modes with const strings?
Instead of firing `onNavigationStateChanged(NavigationState.FreeDrive)` we just `onNavigationStateChanged("FreeDrive")`?

Reply:
I usually don't prefer solutions which involves string matching. Things can go wrong if there are changes to string knowingly/unknowingly. I would rather prefer `onNavigationStarted`, `onFreeDriveStarted` etc and have separate functions for each `NavigationState`. Same for cameramode.

Reply:
Addressed this in my latest update. Resolving this thread.